addons/mail/static/src/models/message_action.js

```diff
@@ -0,0 +1,336 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+import { clear, insertAndReplace, replace } from '@mail/model/model_field_command';
+import { markEventHandled } from '@mail/utils/utils';
+
+registerModel({
+    name: 'MessageAction',
+    identifyingFields: [[
+        'messageActionListOwnerAsDelete',
+        'messageActionListOwnerAsEdit',
+        'messageActionListOwnerAsMarkAsRead',
+        'messageActionListOwnerAsReaction',
+        'messageActionListOwnerAsReplyTo',
+        'messageActionListOwnerAsToggleCompact',
+        'messageActionListOwnerAsToggleStar',
+    ]],
+    recordMethods: {
+        /**
+         * @private
+         * @param {MouseEvent} ev
+         */
+        onClick(ev) {
+            switch (this.messageActionListOwner) {
+                case this.messageActionListOwnerAsDelete:
+                    this.update({ deleteConfirmDialog: insertAndReplace() });
+                    break;
+                case this.messageActionListOwnerAsEdit:
+                    this.messageActionListOwner.messageView.startEditing();
+                    break;
+                case this.messageActionListOwnerAsMarkAsRead:
+                    this.messageActionListOwner.message.markAsRead();
+                    break;
+                case this.messageActionListOwnerAsReaction:
+                    if (this.messageActionView) {
+                        this.messageActionView.openReactionPopver();
+                    }
+                    break;
+                case this.messageActionListOwnerAsReplyTo:
+                    markEventHandled(ev, 'MessageActionList.replyTo');
+                    this.messageActionListOwner.messageView.replyTo();
+                    break;
+                case this.messageActionListOwnerAsToggleCompact:
+                    this.messageActionListOwner.update({ isCompact: !this.messageActionListOwner.isCompact });
+                    break;
+                case this.messageActionListOwnerAsToggleStar:
+                    this.messageActionListOwner.message.toggleStar();
+                    break;
+                default:
+                    break;
+            }
+        },
+        /**
+         * @private
+         * @returns {boolean|FieldCommand}
+         */
+        _computeAriaPressedState() {
+            if (this.messageActionListOwnerAsToggleStar) {
+                if (this.messageActionListOwnerAsToggleStar.message.isStarred) {
+                    return true;
+                }
+                return false;
+            }
+            return clear();
+        },
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeClassAttributes() {
```

Comment:
now that I see it, it would probably be better to have all the "template related" fields in the view model